core/src/main/scala/cats/Parallel.scala

```diff
@@ -0,0 +1,164 @@
+package cats
+
+import cats.arrow.FunctionK
+
+/**
+  * Some types that form a Monad, are also capable of forming an Applicative that supports parallel composition.
+  * The Parallel type class allows us to represent this relationship.
+  */
+trait Parallel[M[_], F[_]] extends Serializable {
+  /**
+    * The applicative instance for F[_]
+    */
+  def applicative: Applicative[F]
+
+  /**
+    * The monad instance for M[_]
+    */
+  def monad: Monad[M]
+
+  /**
+    * Natural Transformation from the parallel Applicative F[_] to the sequential Monad M[_].
+    */
+  def sequential: F ~> M
+
+  /**
+    * Natural Transformation from the sequential Monad M[_] to the parallel Applicative F[_].
+    */
+  def parallel: M ~> F
+
+  /**
+    * Provides an `ApplicativeError[F, E]` instance for any F, that has a `Parallel[M, F]`
+    * and a `MonadError[M, E]` instance.
+    * I.e. if you have a type M[_], that supports parallel composition through type F[_],
+    * then you can get `ApplicativeError[F, E]` from `MonadError[M, E]`.
+    */
+  def applicativeError[E](implicit E: MonadError[M, E]): ApplicativeError[F, E] =
+    Parallel.applicativeError[M, F, E](this, E)
```

Comment:
can we reverse this?

If you make this call the `Parallel.appicativeError` then if I override this method if you call the object version we don't get the override.

Either it should be final here, or the object should call this method.